web/app/components/document/sidebar.ts

```diff
@@ -236,29 +236,32 @@ export default class DocumentSidebarComponent extends Component<DocumentSidebarC
     });
   }
 
-  updateProduct = restartableTask(async (product: string) => {
-    this.product = product;
-    await this.save.perform("product", this.product);
-  });
+  updateProduct = restartableTask(
+    async (product: string) => {
+      this.product = product;
+      await this.save.perform("product", this.product);
+      // productAbbreviation is computed by the back end
+    }
+  );
 
-  save = task(async (field, val) => {
+  save = task(async (field: string, val: string | HermesUser[]) => {
     if (field && val) {
-      let targetField = field;
-      assert("targetField must exist", targetField);
-      const oldVal = targetField;
-      targetField = cleanString(val);
+      let serializedValue;
 
-      try {
-        const serializedValue = this.emailFields.includes(field)
-          ? val.map((p: HermesUser) => p.email)
-          : val;
+      if (typeof val === "string") {
+        serializedValue = cleanString(val);
+      } else {
+        serializedValue = val.map((p: HermesUser) => p.email);
+      }
 
+      try {
         await this.patchDocument.perform({
-          [field]: cleanString(serializedValue),
+          [field]: serializedValue,
         });
       } catch (err) {
         // revert field value on failure
-        targetField = oldVal;
+        (this as any)[field] = val;
+        console.error(err);
```

Comment:
Did you mean to leave this in or was it just for debugging locally?

Reply:
Intentional, but it makes more sense as a flashMessage. PR'd here: https://github.com/hashicorp-forge/hermes/pull/211